banyan/src/forest/read.rs

```diff
@@ -198,6 +198,13 @@ where
                     } else {
                         let index = &branch.children[next_idx];
                         head.next_pos(&self.mode);
+                        // calculate the range in advance
```

Comment:
Maybe put a comment here, that the range needs to be for this child only.

Reply:
This calculates the range to be skipped. I’m not convinced that the range computed above (which was incorrectly used before) actually should be computed in advance — the comment states this but doesn’t explain why. I think it should be computed only when needed, because for branches it computes nonsense unless in the `head.filter.is_empty()` case (`offset` has already advanced and thus doesn’t fit with `head.index` anymore).